File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.js
function solution(n) {
    var acc = 0;
    if (n == 1)
        return 1;
    if (Math.floor(n % 2) === 1)
        acc++;
    
    while (n !== 1) {
        if (Math.floor(n / 2) % 2 === 1)
            acc++;
        n = Math.floor(n / 2);
    }
    
    return acc;
}

File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.cpp
#include <iostream>
using namespace std;

int solution(int n)
{
    int ans = 1;

    // 이진법에서 1의 수가 총 점프 수.
    while (n / 2)
    {
        if (n % 2) // 홀수일 경우
        {
            --n;
            ++ans;
        }
        n /= 2;
    }

    return ans;
}